ledfx/virtuals.py

```diff
@@ -603,57 +605,6 @@ def _fire_update_event(self, frame=None):
             )
         )
 
-    def oneshot(self, color, ramp, hold, fade):
-        """
-        Force all pixels in virtual to color over a time envelope defined in ms
-        Following calls will override any active one shot
-
-        Parameters:
-            ramp time from in ms
-            hold time in ms
-            fade time in ms
-        Returns:
-            True if oneshot was activated, False if not
-        """
-        if self.active:
-            self._os_color = np.array(color, dtype=float)
-            self._os_ramp = ramp / 1000.0
-            self._os_hold = hold / 1000.0
-            self._os_fade = fade / 1000.0
-            self._os_start = timeit.default_timer()
-            self._os_hold_end = self._os_ramp + self._os_hold
-            self._os_fade_end = self._os_ramp + self._os_hold + self._os_fade
-            self._os_weight = 0.0
-
-            # if all total timings are zero, treat as a hard off
-            if self._os_fade_end == 0:
```

Comment:
This hard off code seems to be missing now in the reworked version, is that intentional?

Reply:
Tested on PR, no hard off behaviour.

Tested on main, hard off works 

I will raise a PR against my very basic test script

Reply:
Hey @bigredfrog, sorry for a late reply. I see what you mean!

I see how this can be cleanly address in the next PR. Coming soon!